include/zephyr/drivers/i2c.h

```diff
@@ -722,6 +722,27 @@ static inline int z_impl_i2c_configure(const struct device *dev,
 	return api->configure(dev, dev_config);
 }
 
+/**
+ * @brief Configure operation of a host controller.
+ *
+ * This is equivalent to:
+ *
+ *     i2c_configure(spec->bus, dev_config);
+ *
+ * @param spec I2C specification from devicetree.
+ * @param dev_config Bit-packed 32-bit value to the device runtime configuration
+ * for the I2C controller.
+ *
+ * @return a value from i2c_configure()
+ */
+__syscall int i2c_configure_dt(const struct i2c_dt_spec *spec, uint32_t dev_config);
```

Comment:
Doesn't need to be a syscall, can be a simple inline wrapper around i2c_configure, otherwise looks ok to me